cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -507,6 +511,13 @@ private ProgramOptions createProgramOptions(File programOptionsFile) throws IOEx
         addOnPremiseServices(injector, programOptions, metricsCollectionService, services);
         break;
       case ISOLATED:
+        try {
+          // Define all StructuredTable before starting any services that need StructuredTable
+          StoreDefinition.createAllTables(injector.getInstance(StructuredTableAdmin.class),
+                  injector.getInstance(StructuredTableRegistry.class));
+        } catch (IOException | TableAlreadyExistsException e) {
+          throw new RuntimeException("Unable to create the system tables.", e);
+        }
```

Comment:
I though we only need to create the tables used by dataset service? Why create all of them?

Reply:
That would probably be just these 2, then?

```java
    if (overWrite || tableAdmin.getSpecification(DatasetInstanceStore.DATASET_INSTANCES) == null) {
      DatasetInstanceStore.createTables(tableAdmin);
    }
    if (overWrite || tableAdmin.getSpecification(DatasetTypeStore.DATASET_TYPES) == null) {
      DatasetTypeStore.createTables(tableAdmin);
    }
```

Reply:
yes